server/protocols/jmap-rfc-8621/src/test/scala/org/apache/james/jmap/change/MailboxChangeListenerTest.scala

```diff
@@ -126,6 +127,23 @@ class MailboxChangeListenerTest {
       .containsExactly(inboxId)
   }
 
+  @Test
+  def updateMessageFlagsShouldStoreUpdateEvent(): Unit = {
+    val mailboxSession = MailboxSessionUtil.create(BOB)
+    val path = MailboxPath.forUser(BOB, "test")
+    val inboxId: MailboxId = mailboxManager.createMailbox(path, mailboxSession).get
+    val messageManager: MessageManager = mailboxManager.getMailbox(inboxId, mailboxSession)
+    messageManager.appendMessage(AppendCommand.builder().build("header: value\r\n\r\nbody"), mailboxSession)
+
+    val state = State.of(UUID.randomUUID)
+    repository.save(MailboxChange.of(ACCOUNT_ID, state, ZonedDateTime.now, List[MailboxId](TestId.of(0)).asJava, List().asJava, List().asJava)).block()
+
+    messageManager.setFlags(new Flags(Flags.Flag.SEEN), FlagsUpdateMode.ADD, MessageRange.all(), mailboxSession)
+
+    assertThat(repository.getSinceState(ACCOUNT_ID, state, None.toJava).block().getUpdated)
+      .containsExactly(inboxId)
+  }
+
```

Comment:
Add a test with a flag update test without the SEEN flag, it should not update things!